cli/command/environment.go

```diff
@@ -217,42 +231,52 @@ func (e *EnvironmentCommand) update(c *cli.Context) error {
 }
 
 func (e *EnvironmentCommand) link(c *cli.Context) error {
-	args, err := extractArgs(c.Args(), "SOURCE_ENVIRONMENT_NAME", "DESTINATION_ENVIRONMENT_NAME")
-	if err != nil {
-		return err
-	}
+	generateAddLinkRequest := func(sourceEnvironmentID, destEnvironmentID string) (models.UpdateEnvironmentRequest, error) {
+		env, err := e.client.ReadEnvironment(sourceEnvironmentID)
+		if err != nil {
+			return models.UpdateEnvironmentRequest{}, err
+		}
 
-	sourceEnvironmentID, err := e.resolveSingleEntityIDHelper("environment", args["SOURCE_ENVIRONMENT_NAME"])
-	if err != nil {
-		return err
-	}
+		req := models.UpdateEnvironmentRequest{EnvironmentID: sourceEnvironmentID}
+		appendedLinks := append(env.Links, destEnvironmentID)
+		req.Links = &appendedLinks
 
-	destEnvironmentID, err := e.resolveSingleEntityIDHelper("environment", args["DESTINATION_ENVIRONMENT_NAME"])
-	if err != nil {
-		return err
+		return req, nil
 	}
 
-	if sourceEnvironmentID == destEnvironmentID {
-		return fmt.Errorf("Cannot link an environment to itself")
-	}
+	return e.updateEnvironmentLinksHelper(c, generateAddLinkRequest)
+}
 
-	req := models.CreateEnvironmentLinkRequest{
-		SourceEnvironmentID: sourceEnvironmentID,
-		DestEnvironmentID:   destEnvironmentID,
-	}
+func (e *EnvironmentCommand) unlink(c *cli.Context) error {
+	generateRemoveLinkRequest := func(sourceEnvironmentID, destEnvironmentID string) (models.UpdateEnvironmentRequest, error) {
+		env, err := e.client.ReadEnvironment(sourceEnvironmentID)
```

Comment:
my old example didn't require the helper call `ReadEnvironment`

Reply:
ok - not sure if you're asking me to copy your example verbatim? The change to refactor link and unlink in the command and provider layer were underway and I referenced your example on the WIP PR but I didn't copy it.

Reply:
It's doesn't have to be verbatim, but it does save you the read call in the "generateRequest" helpers